gnovm/pkg/gnolang/type_check.go

```diff
@@ -288,6 +288,9 @@ func checkAssignableTo(xt, dt Type, autoNative bool) error {
 	}
 	// case0
 	if xt == nil { // see test/files/types/eql_0f18
+		if dt == nil || dt.Kind() == InterfaceKind {
```

Comment:
Can you provide a test showing why the second half of this conditional is necessary?

Reply:
Added. please check the `type_check_test.go` file https://github.com/gnolang/gno/pull/2699/commits/e9c055de035d9feada6eeeba4a929685764ed647#diff-ed2a83286fe8c8215a33b8569bb74791d51ae4e0800ef81c326c55f16a85123c